### PR TITLE
TRAN-76: added Searcher, redux for CustomerType and simple skeleton for CustomerType page

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://sd-transit.atlassian.net/browse/TRAN-76

Presentation how filtering works (table will be delivered as a part of https://sd-transit.atlassian.net/browse/TRAN-74). 
Peek 2022-08-31 13-57

Including minor changes in refreshing token visible on SignInForm (after refreshing token while switching page url user will go back to that page, not to landing page)

Changes:
- added Searcher with prop (refetch) - this prop is the any function defined at the level of the given page responsible for fetching data from reducer based on context from 'searcher' value.
- added redux for CustomerType. The token refresh is added for request with 'Bearer Token'
- added simple skeleton for CustomerType page to demonstrate filtering by using Searcher compoment. 
- installed 'react-icons' library
- added new 'Clear' button to clear forms. 
- added new page to 'routing' (as a 'protected route')

### CODE REVIEW

File: src/pages/menuUpload/CustomerTypePage.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <div>
+      <div>
+        <Searcher refetch={refetch} />
+      </div>
+      <div>
+        {customerTypes === undefined ? (
+          <div />
+        ) : (
+          customerTypes.map((customer_type: ICustomerType) => (
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/CustomerTypePage.tsx]: Try just 
       ```
 {customerTypes &&
          customerTypes.map......
```
:)

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

File: src/shared/images/SearchIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+function SearchIcon() {
```
Comment (awalkowiak) [inline src/shared/images/SearchIcon.tsx]: @sniedzielski @KarolinaKopacz would it not be better to use a library with icons? You could use this one https://react-icons.github.io/react-icons/ which gathers the most common icon libs.

Comment (sniedzielski) [reply src/shared/images/SearchIcon.tsx]: @awalkowiak I installed lib 'react-lib' in that commit. https://github.com/SD-TransIT/transit_frontend/pull/21/commits/10caf9757ae6315c3ee67e8b29baf299bee03fdb. This commit also includes taking <FiSearch> icon from that lib.

File: src/components/shared/Searcher.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          />
+        </div>
+        <div className="inline-block w-1/12">
+          <ClearButton
+            onClick={() => {
+              reset({
+                search: '',
+              });
+            }}
+            title="Clear"
```
Comment (KarolinaKopacz) [inline src/components/shared/Searcher.tsx]: ClearButton already has title = 'Clear" so this props here is unnecessary :)

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: src/components/shared/Searcher.tsx (hunk 3 above)
Comment (sniedzielski) [reply src/components/shared/Searcher.tsx]: fixed

**Review by KarolinaKopacz — APPROVED** [review]

**Pull request merged**